src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,41 @@
+package mate.academy.service.impl;
+
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.AuthenticationService;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        User byEmail = userService.findByEmail(email).orElseThrow(() ->
+                new AuthenticationException("User with : <" + email + "> and <"
+                        + password + "> don't passed authentication process"));
+        if (!byEmail.getPassword()
+                .equals(HashUtil.hashPassword(password, byEmail.getSalt()))) {
+            throw new AuthenticationException(" For User with ID <" + byEmail.getId()
+                    + "> password was inputting incorrectly");
+        }
+        return byEmail;
+    }
+
+    @Override
+    public User register(String email, String password) throws RegistrationException {
+        if (email == null || password == null || email.isEmpty() || password.isEmpty()) {
```

Comment:
According the previous review comment this `if` can be simplified with using StringUtils.isEmpty() or
[StringUtils.isBlank()](http://commons.apache.org/proper/commons-lang/javadocs/api-2.6/org/apache/commons/lang/StringUtils.html#isBlank%28java.lang.String%29)

Reply:
refactored 
StringUtils.isEmpty() is prefered